dicthash/dicthash.py

```diff
@@ -13,10 +13,14 @@
 
 """
 
+from future.builtins import str
 import hashlib
-
 FLOAT_FACTOR = 1e15
 
+try:
```

Comment:
So, this is essentially the same problem that we came across in the h5py_wrapper.
Here is the explanation:
- All strings are now unicode, following Python 3 policy.
- To enable this with one common code base, we import the `str` Data class from future.
- In python 2, this leads to a problem if a string is defined as a python 2 string, because there is then no more way to check its type, since the `str` function has been overloaded with the new Python 3 data class.
- To remedy this, we check for `basestring` instead of checking for `str`, since this code
     `s = 'a'  # a python2 string `
    ` from future.builtins import str`
    ` isinstance(s, str)`
  `   >>> False`
- Since `basestring` is only defined in Python 2, we have to define it in Python 3. The `try` statement checks whether `basestring` is defined and if not, defines it as `str` (in py3).

The idea is taken from here [http://stackoverflow.com/questions/11301138/how-to-check-if-variable-is-string-with-python-2-and-3-compatibility](http://stackoverflow.com/questions/11301138/how-to-check-if-variable-is-string-with-python-2-and-3-compatibility)